### ISSUE TITLE
Multi image ASRP datasets open using the GEN file are incorrectly identified as ADRG

### ISSUE THREAD

**Issue by sab56** [OSGeo/gdal#953]
## Problem Description.

When the OPEN_GEN option is enabled and an ASRP dataset containing multiple images is opened using the GEN file the dataset is incorrectly identified as ADRG.

## Steps to reproduce the problem.

```
C:\>gdalinfo TPCE1B01.GEN
Driver: ADRG/ARC Digitized Raster Graphics
Files: TPCE1B01.GEN
Size is 512, 512
Coordinate System is:
GEOGCS["WGS 84",
    DATUM["WGS_1984",
        SPHEROID["WGS 84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["degree",0.0174532925199433],
    AUTHORITY["EPSG","4326"]]
Subdatasets:
  SUBDATASET_1_NAME=ADRG:TPCE1B01.GEN,.\TPCE1B01.IMG
  SUBDATASET_1_DESC=ADRG:TPCE1B01.GEN,.\TPCE1B01.IMG
  SUBDATASET_2_NAME=ADRG:TPCE1B01.GEN,.\TPCE1B02.IMG
  SUBDATASET_2_DESC=ADRG:TPCE1B01.GEN,.\TPCE1B02.IMG
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,  512.0)
Upper Right (  512.0,    0.0)
Lower Right (  512.0,  512.0)
Center      (  256.0,  256.0)

```

Simple ASRP dataset containing 2 arc zone images:
[TPCE1B01.zip](https://github.com/OSGeo/gdal/files/2427806/TPCE1B01.zip)

## Suggested Solution

Modify the ADRG driver to differentiate between ADRG and ASRP datasets by checking for the existence of a NWO subfield in the GEN field and returning a null pointer if it is not found (SRP datasets do not contain the NWO subfield). This can be done by adding the following code to the function ADRGDataset::GetIMGListFromGEN, immediately after the statement "if ( strcmp(RTY, "GIN") != 0 ) continue;":

```
/* test for existence of a NWO subfield in the GEN field */
            const char* NWO = record->GetStringSubfield("GEN", 0, "NWO", 0);
            if( NWO == nullptr )
                return nullptr;

```
## Operating system

Windows 7 and 10

## GDAL version and provenance

GDAL 1.11.1 and later

**Comment by rouault** [comment]
Do you wish to submit your proposed change as a pull request ? For the sake of correctness (I don't think this is necessary for normal datasets, but could possibly happen on corrupted ones), a CSLDestroy(papszFileNames) should be done before returning nullptr

**Comment by sab56** [comment]
Hi Even,

I’m new to GitHub and don’t know what the term ‘pull request’ means.

There is another issue regarding the opening of ASRP datasets using the THF that I was about to submit.

Kind regards,
Steve Baker

**Comment by rouault** [comment]
> I’m new to GitHub and don’t know what the term ‘pull request’ means.

A pull request is a branch with changes you push to the fork on your github account, and submit for inclusion to the parent repository. GDAL uses a standard workflow, which is briefly described at https://github.com/OSGeo/gdal/blob/master/CONTRIBUTING.md, but you may need to explore dedicated tutorials to get more familiar

**Comment by sab56** [comment]
Thanks, I'll give it a go.
